activesupport/test/core_ext/string_ext_test.rb

```diff
@@ -249,6 +249,19 @@ def test_truncate_words_with_separator_and_omission
     assert_equal "Hello<br>Big<br>World!", "Hello<br>Big<br>World!".truncate_words(3, :omission => "[...]", :separator => '<br>')
   end
 
+  def test_truncate_words_with_complex_string
+    begin
+      assert_nothing_raised do
+        Timeout.timeout(10) do
+          complex_string = "aa aa aaa aa aaa aaa aaa aa aaa aaa aaa aaa aaa aaa aaa aaa aaa aaa aaaa aaaaa aaaaa aaaaaa aa aa aa aaa aa  aaa aa aa aa aa a aaa aaa \n a aaa <<s"
+          assert_equal complex_string.truncate_words(80), complex_string
+        end
+      end
+    rescue Timeout::Error
```

Comment:
Is this rescue necessary? I think `assert_nothing_raised` will already handle this case.